### DISCUSSION TITLE
ERROR: failed to calculate checksum of ref 2be76fa3-2818-4c9b-915a-8fa715bd0b82::lhij89b8k6fneaah4jc4kjrg2: "/jenkins-agent": not found

### DISCUSSION THREAD

**Discussion by moulikatusam** [docker/build-push-action#935] [Q&A]
![image (19)](https://github.com/docker/build-push-action/assets/136565076/963b4ebd-414d-49af-9fb3-61f08aa7c97a)
ERROR: failed to calculate checksum of ref 2be76fa3-2818-4c9b-915a-8fa715bd0b82::lhij89b8k6fneaah4jc4kjrg2: "/jenkins-agent": not found

I am getting the above error for the following code. I have tried different ways for the COPY line but none worked.

USER root
RUN ls -al
COPY ./jenkins-agent /usr/local/bin/jenkins-agent
RUN chmod +x /usr/local/bin/jenkins-agent &&
ln -s /usr/local/bin/jenkins-agent /usr/local/bin/jenkins-slave

Expected behaviour
expecting this  COPY ./jenkins-agent /usr/local/bin/jenkins-agent to pass with no error as follows
failed to calculate checksum of ref 2be76fa3-2818-4c9b-915a-8fa715bd0b82::lhij89b8k6fneaah4jc4kjrg2: "/jenkins-agent": not found.

The Jenkins-agent file and Dockerfile both are in/under same directory. Yet it is throwing the above error. Unable to understand what is wrong with the COPY  command.

Actual behavior
10:54:53 https://github.com/docker/build-push-action/issues/32 ERROR: failed to calculate checksum of ref 2be76fa3-2818-4c9b-915a-8fa715bd0b82::lhij89b8k6fneaah4jc4kjrg2: "/jenkins-agent": not found
10:54:53 https://github.com/docker/build-push-action/pull/8 extracting sha256:de6a43bfad433fa7a8b0c1b364fe2bfb9cdd6b8315a9a71e5252e46c0d00d039 1.1s done
10:54:53 ------
10:54:53 > [24/28] COPY ./jenkins-agent /usr/local/bin/jenkins-agent:
10:54:53 ------
10:54:53 Dockerfile:100
10:54:53 --------------------
10:54:53 98 | USER root
10:54:53 99 | RUN ls -al
10:54:53 100 | >>> COPY ./jenkins-agent /usr/local/bin/jenkins-agent
10:54:53 101 | RUN chmod +x /usr/local/bin/jenkins-agent &&
10:54:53 102 | ln -s /usr/local/bin/jenkins-agent /usr/local/bin/jenkins-slave
10:54:53 --------------------
10:54:53 ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref 2be76fa3-2818-4c9b-915a-8fa715bd0b82::lhij89b8k6fneaah4jc4kjrg2: "/jenkins-agent": not found

**Comment by IamTheFij** [comment]
Did you ever figure this out? I'm seeing a similar issue myself.

**Reply by IamTheFij** [reply]
Ah. Answer is here: https://github.com/docker/build-push-action?tab=readme-ov-file#git-context

tl;dr - It doesn't use the path context but the git context. Gotta add `context: .` so it can access local files that you add in previous steps.
